src/plugins/oSnap/components/Input/MethodParameter.vue

```diff
@@ -17,34 +21,93 @@ const emit = defineEmits<{
 }>();
 
 const isDirty = ref(false);
-const isBooleanInput = computed(() => props.parameter.baseType === 'bool');
-const isAddressInput = computed(() => props.parameter.baseType === 'address');
-const isNumberInput = computed(() => props.parameter.baseType.includes('int'));
-const isBytesInput = computed(() => props.parameter.baseType === 'bytes');
-const isBytes32Input = computed(() => props.parameter.baseType === 'bytes32');
-const isArrayInput = computed(
+
+const placeholders = {
+  string: 'a string of text',
+  address: '0x123...abc',
+  int: '123',
+  bytes: '0x123abc',
+  bytes32: '0x123abc',
+  bool: 'true'
+} as const;
+
+const inputType = computed(() => {
+  const baseType = props.parameter.baseType;
+
+  if (baseType === 'tuple') {
+    return {
+      input: 'tuple',
+      type: props.parameter.components.map(item => item.baseType as InputTypes)
+      // ["string","int","address"]
+    } as const;
+  }
+
+  if (baseType === 'array') {
+    return {
+      input: 'array',
+      type: props.parameter.arrayChildren.baseType as InputTypes
+    } as const;
+  }
+
+  return { type: baseType as InputTypes, input: 'single' } as const;
+});
+
+const isBooleanInput = computed(
+  () => inputType.value.input === 'single' && inputType.value.type === 'bool'
+);
+const isStringInput = computed(
+  () => inputType.value.input === 'single' && inputType.value.type === 'string'
+);
+const isAddressInput = computed(
+  () => inputType.value.input === 'single' && inputType.value.type === 'address'
+);
+const isNumberInput = computed(
   () =>
-    props.parameter.baseType === 'array' || props.parameter.baseType === 'tuple'
+    inputType.value.input === 'single' && inputType.value.type.includes('int')
+);
+const isBytesInput = computed(
+  () => inputType.value.input === 'single' && inputType.value.type === 'bytes'
+);
+const isBytes32Input = computed(
+  () => inputType.value.input === 'single' && inputType.value.type === 'bytes32'
 );
+const isArrayInput = computed(() => inputType.value.input !== 'single');
 
-const inputType = computed(() => {
-  if (isBooleanInput.value) return 'boolean';
-  if (isAddressInput.value) return 'address';
-  if (isNumberInput.value) return 'number';
-  if (isBytesInput.value) return 'bytes';
-  if (isBytes32Input.value) return 'bytes32';
-  if (isArrayInput.value) return 'array';
-  return 'text';
+// param name may be null or empty string
+const paramName = `${
+  props.parameter.name?.length ? props.parameter.name + ' ' : ''
+}`;
+const paramType = computed(() => {
+  if (inputType.value.input === 'single') {
+    return `(${inputType.value.type})`;
+  }
+  if (inputType.value.input === 'array') {
+    return `(${inputType.value.type}[])`;
+  }
+  // tuple type labels can be too long and take up too much space, limit to 2
+  return inputType.value.type.length > 2
+    ? `( ${inputType.value.type.slice(0, 2)}...[ ] )`
+    : inputType.value.type;
+});
+
+const label = paramName + paramType.value;
```

Comment:
This logic is really just to make sure the label for the input isn't too long since tuple types can be complex:

![Screenshot 2024-04-12 at 16 24 38](https://github.com/UMAprotocol/snapshot/assets/51655063/2223dfeb-18f2-4fb0-b15a-738ab5c25b53)